sql/executor.go

```diff
@@ -678,8 +679,10 @@ func checkResultDatum(datum parser.Datum) error {
 	case parser.DDate:
 	case parser.DTimestamp:
 	case parser.DInterval:
+	case parser.DValArg:
+		return fmt.Errorf("could not infer %s", datum.Type())
```

Comment:
I think this should be `return fmt.Errorf("could not determine data type of parameter %s", d.name)` with `switch d := datum.(type) {` at the top. This matches postgres' error message.